File: frontend/src/api/species.ts
import { PageResponse, PaginationParams, get, post, put } from "./base";

export interface SearchParams {
  level: string;
  keyword?: string;
  pagination: PaginationParams;
}

export interface SearchResultItem {
  briefIntroduction?: string;
  genusId: number;
  cName: string;
  createTime?: string;
  latinName: string;
  detailIntroduction?: string;
  id: number;
  pictureUrl?: string;
  betterUrl?: string;
  recommend: boolean;
  updateTime: string;
  categorizedInheritance: string;
  inheritance: Record<"1" | "2" | "3" | "4" | "5", Array<string>>;
}

export const searchSpecies = (data: SearchParams) => {
  return get<PageResponse<SearchResultItem>>(`/api/species/blur`, {
    inheritance: data.level,
    keyword: data.keyword,
    ...data.pagination,
  });
};

export const getSpecies = (id: number) => {
  return get<SearchResultItem>(`/api/species/${id}`);
};

export const updateSpecies = (id: number, data: any) => {
  return put(`/api/species/${id}`, data);
};

export const createSpecies = (data: any) => {
  data.id = -1;
  return post(`/api/species`, data);
};

export const recommendSpecies = () => {
  return get<Array<SearchResultItem>>("/api/species/recommends");
};

File: backend/src/main/java/com/neutech/mammalia/bean/Species.java
package com.neutech.mammalia.bean;

import java.util.Date;
import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 物种
 * t_species
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Species {
    /**
     * 主键
     */
    private Integer id;
    /**
     * 物种的中文名称
     */
    @JsonProperty
    private String cName;
    /**
     * 物种的拉丁文正式名
     */
    private String latinName;
    /**
     * 该物种所属的类别
     */
    private Integer genusId;
    /**
     * 物种简介
     */
    private String briefIntroduction;
    /**
     * 物种详情
     */
    private String detailIntroduction;
    /**
     * 是否推荐.true:推荐false:不推荐.每日24点重新生成推荐内容
     */
    private Boolean recommend;
    /**
     * 保护级别. 1:国1; 2:国2; 3:三有; 0:非保护动物;
     */
    private Integer level;
    /**
     * 创建时间
     */
    private Date createTime;
    /**
     * 上次更新时间
     */
    private Date updateTime;
    /**
     * 物种照片地址
     */
    private String pictureUrl;
    /**
     * better
     */
    private String betterUrl;
    /**
     * 继承关系
     */
    private Map<Integer, List<String>> inheritance;
}
